src/app/reporting/SynchronizedReportSchedulerImpl.cpp

```diff
@@ -138,7 +133,10 @@ CHIP_ERROR SynchronizedReportSchedulerImpl::CalculateNextReportTimeout(Timeout &
     bool reportableNow   = false;
     bool reportableAtMin = false;
 
+    // Find out if any handler is reportable now or at the next min interval
     mNodesPool.ForEachActiveObject([&reportableNow, &reportableAtMin, this, now](ReadHandlerNode * node) {
+        // If a node is already scheduled, we don't need to check if it is reportable now, unless a chunked report is in progress
+        // in which case we need to keep scheduling engine runs until the report is complete
```

Comment:
Yes, node->IsChunkedReport() can only be true while reporting, which only happens on reportable handler.
Also, ReadHandler.h features a method:
`bool IsReporting() const { return mFlags.Has(ReadHandlerFlags::ChunkedReport); }`
That checks the same thing as IsChunkedReport(), I was considering this method redundant but since it's more clear that we are in the middle of a report, maybe it would be an idea to use it instead?